web-api/tests/filter.rs

```diff
@@ -391,3 +391,162 @@ fn test_filter_array_string_multiple() {
+                    }))
+                    .build()?,
+                StatusCode::OK,
+            )
+            .await;
+            assert_eq!(documents.ids(), ["d2"].into());
+
+            let documents = send_assert_json::<SemanticSearchResponse>(
+                &client,
+                client
+                    .post(personalization_url.join("/semantic_search")?)
+                    .json(&json!({
+                        "document": { "query": "zero" },
+                        "filter": {
+                            "$and": [{ "p1": { "$eq": "this" } }, { "p2": { "$eq": "too" } }]
+                        }
+                    }))
+                    .build()?,
+                StatusCode::OK,
+            )
+            .await;
+            assert!(documents.is_empty());
+
+            let documents = send_assert_json::<SemanticSearchResponse>(
+                &client,
+                client
+                    .post(personalization_url.join("/semantic_search")?)
+                    .json(&json!({
+                        "document": { "query": "zero" },
+                        "filter": {
+                            "$or": [{ "p1": { "$eq": "that" } }, { "p2": { "$eq": "word" } }]
+                        }
+                    }))
+                    .build()?,
+                StatusCode::OK,
+            )
+            .await;
+            assert_eq!(documents.ids(), ["d2", "d3"].into());
+
+            let documents = send_assert_json::<SemanticSearchResponse>(
+                &client,
+                client
+                    .post(personalization_url.join("/semantic_search")?)
+                    .json(&json!({
+                        "document": { "query": "zero" },
+                        "filter": {
+                            "$or": [{ "p1": { "$eq": "foo" } }, { "p2": { "$eq": "bar" } }]
+                        }
+                    }))
+                    .build()?,
+                StatusCode::OK,
+            )
+            .await;
+            assert!(documents.is_empty());
+
+            let documents = send_assert_json::<SemanticSearchResponse>(
+                &client,
+                client
+                    .post(personalization_url.join("/semantic_search")?)
+                    .json(&json!({ "document": { "query": "zero" }, "filter": { "$and": [] } }))
+                    .build()?,
+                StatusCode::OK,
+            )
+            .await;
+            assert_eq!(documents.ids(), ["d1", "d2", "d3"].into());
+
+            let documents = send_assert_json::<SemanticSearchResponse>(
+                &client,
+                client
+                    .post(personalization_url.join("/semantic_search")?)
+                    .json(&json!({ "document": { "query": "zero" }, "filter": { "$or": [] } }))
+                    .build()?,
+                StatusCode::OK,
+            )
+            .await;
+            assert_eq!(documents.ids(), ["d1", "d2", "d3"].into());
```

Comment:
:wrench:  There is a bug somewhere here I think.

`"$or": []` should always be false, i.e. not match any documents

I mean it should map to `{ "should":  [], "minimum_should_match": 1 }` which can never evaluate to have enough matches :thinking: 

(Also logically `$and: [ (...),*]` is like `true (&& ...)*` and `$or: [...,*]` like `false (|| ...)*` )

Reply:
this is forwarded as is to es, so we probably have to decide if es's edge case handling is what we want or not.

Reply:
ES :face_exhaling: 

I guess that becomes a [:pushpin: issue](https://xainag.atlassian.net/browse/ET-4738).